lib/vintage_net_qmi/connection.ex

```diff
@@ -128,6 +128,11 @@ defmodule VintageNetQMI.Connection do
 
         state
 
+      {:error, :no_effect} ->
+        # no effect means that a network connection as already be established
+        # so we don't need to try to connect again.
+        state
```

Comment:
Shouldn't this set the APN as well?

Reply:
Oh, I think that reporting the chosen APN needs to be moved up to right before `WirelessData.start_network_interface`. I.e., if the network interface doesn't start, we'd want to know what APN was being tried.

Reply:
@fhunleth updated!